bitcoin/src/address.rs

```diff
@@ -158,6 +158,16 @@ impl From<bech32::Error> for Error {
     fn from(e: bech32::Error) -> Error { Error::Bech32(e) }
 }
 
+#[doc(hidden)]
+impl From<witness_version::Error> for Error {
+    fn from(e: witness_version::Error) -> Error { Error::WitnessVersion(e) }
+}
+
+#[doc(hidden)]
```

Comment:
Why are these hidden? I think they shouldn't be.

Reply:
We've always had `doc(hidden)` on `From` impls that are intended to be used via `?`. IIRC this is what `thiserror` does as well.

Reply:
I copied the others in this file but I have not been doing this (or don't remember seeing it) in any other files. Do we want this on all error `From`s?

Reply:
IDK what `thiserror` does but I consider this a bad idea. `hidden` implies it shouldn't be used by downstream code but it's super-easy to use it by accident. So we should either commit to supporting these conversions and un-hide them or remove them.

Reply:
> `hidden` implies it shouldn't be used by downstream code

No, it's `hidden` because it's basically just noise. The intent is that users will use it via `?` and never know/care that there is a conversion happening.

Reply:
Yeah that's the understanding I had too,  especially with the new API checking script we are definitely committing to these implementations in my mind.

Reply:
I don't remember seeing any use of `hidden` other than hiding non-public items that have to be `pub`. And if we want them to be public why not document the conversions? Most people don't read the code but documentation. If the conversion is not documented it looks like missing.

I don't think it's noise - it's not always obvious that the conversion is available and it's also documented around the bottom of an error type. It doesn't get in the way of other, more important things.

Reply:
>Most people don't read the code but documentation

I don't think people do either. I think they stick `?` onto their errors and then check for resulting compiler errors, if any, to determine whether they manually need to convert error types. Only then would they look at the docs.

I'm happy to remove the `doc(hidden)` on these conversions but it's a pretty common convention, or at least it used to be, for conversions that only exist to make `?` work.

Reply:
Ok, that's reasonable. Let's un-hide these.

Reply:
Just to check my reading of the discussion; resolution is to never use hidden on error type From impls, correct?

Reply:
@tcharding correct.

Reply:
Awesome, done in #1887